plugin/evm/atomic/sync/atomic_sync_parser.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package sync
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/coreth/plugin/evm/message"
+	"github.com/ava-labs/libevm/crypto"
+)
+
+var _ message.SyncableParser = (*atomicSyncSummaryParser)(nil)
+
+type atomicSyncSummaryParser struct{}
+
+func NewAtomicSyncSummaryParser() *atomicSyncSummaryParser {
+	return &atomicSyncSummaryParser{}
+}
+
+func (a *atomicSyncSummaryParser) ParseFromBytes(summaryBytes []byte, acceptImpl message.AcceptImplFn) (message.Syncable, error) {
+	summary := AtomicSyncSummary{}
+	if codecVersion, err := Codec.Unmarshal(summaryBytes, &summary); err != nil {
+		return nil, fmt.Errorf("failed to parse syncable summary: %w", err)
+	} else if codecVersion != message.Version {
+		return nil, fmt.Errorf("failed to parse syncable summary due to unexpected codec version (got %d, expected %d)", codecVersion, message.Version)
+	}
```

Comment:
Because we only register a single codec version on this codec, there is no reason to check the version. It is already guaranteed to be `message.Version`
```suggestion
	if _, err := Codec.Unmarshal(summaryBytes, &summary); err != nil {
		return nil, fmt.Errorf("failed to parse syncable summary: %w", err)
	}
```